src/monodroid/jni/basic-utilities.hh

```diff
@@ -154,13 +148,11 @@ namespace xamarin::android
 			return strdup_new (s, strlen (s));
 		}
 
-		// Without <type_traits> it's a little bit open for abuse (bad stuff will happen if
-		// a type different than `char*` is used to specialize the function and we can't
-		// assert this condition on compile time), but we can take that risk since it's
-		// internal, controlled API
-		template<typename StringType = const char*, typename ...Strings>
-		char* string_concat (const char *s1, StringType s2, Strings... strings)
+		template<typename StringType = char, typename ...Strings>
```

Comment:
I do wonder if we're "prematurely designing" here, introducing template `StringType` parameter types (`CharType` parameter types? see other comment), when the *only* type this will *currently* be is `char`.  Why do we need the complexity of generic type parameters *now*?

Reply:
I want to use `char8_t` for UTF-8 strings once NDK updates to a version of clang that supports it. At the same time, we will STILL need to be able to use `char` for the string pointers we pass to Mono (for simplicity, as `char8_t` cannot be just simply cast to `char`)